java/sage/EPG.java

```diff
@@ -624,19 +626,37 @@ public void run()
     // JS 8/21/2016: This is no longer used.
     //boolean[] didAdd = new boolean[1];
     boolean updateFinished = true;
+    //JUSJOKEN: 2025-02-19 add ability to run FULL Maintenance at a specific time each day
     while (alive)
     {
+
       try{
 
-        if (Sage.time() - wiz.getLastMaintenance() > MAINTENANCE_FREQ)
-          reqMaintenanceType = MaintenanceType.FULL;
+        if(Sage.getBoolean("wizard/scheduled_maintenance", false)){
+            if (Sage.DBG) System.out.println("EPG next scheduled maintenance offset is " + Sage.getInt("wizard/" + SCHEDULED_MAINTENANCE_OFFSET, 0));
+
+            nextScheuldedMaintenanceTime = getNextScheduledMaintenanceTime();
+
+            if ((Sage.time() - wiz.getLastMaintenance() > MAINTENANCE_FREQ) || ((nextScheuldedMaintenanceTime - MAINTENANCE_FREQ) < Sage.time())){
+                if (Sage.DBG) System.out.println("EPG next scheduled update is ready to run as we are past the maintenance window and/or scheduled time");
+                reqMaintenanceType = MaintenanceType.FULL;
+            }else if(wiz.getLastMaintenance()==0){
```

Comment:
I don't think you need this 'else if' block here, because of wiz.getLastMaintenance() is zero...then the first part of the 'if' would be triggered...so this block will never be executed.

Reply:
Agreed.  Removed (I believe I added while testing so I could force an update on restart by setting it to 0 but should have removed this).